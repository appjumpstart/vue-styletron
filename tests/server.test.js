/**
 * @jest-environment node
 */
import { readFileSync } from 'fs'
import { join } from 'path'

import Vue from 'vue/dist/vue'
import { Server } from 'styletron-engine-atomic'
import { createRenderer } from 'vue-server-renderer'

import HelloMessage from './fixtures/HelloMessage.vue'
import VueStyletron from '../'
import renderComponents from '../renderComponents'

// Create a Styletron server instance.
const styletron = new Server()

//
Vue.use(VueStyletron, { styletron })

// Create a renderer instance with the test template.
const template = readFileSync(join(__dirname, './fixtures/index.html'), 'utf-8')
const renderer = createRenderer({ template })

test('server generates styles and classes', async () => {
  try {
    // Create and mount the app with styletron as an option.
    const app = new Vue({ render: h => h(HelloMessage) })

    //
    renderComponents(styletron, )

    // Extract the stylesheet HTML from Styletron.
    // styletron.renderStyle({ backgroundColor: '#000' })
    const styles = styletron.getStylesheetsHtml()

    // Generate the page HTML from the renderer and stylesheet HTML.
    const html = await renderer.renderToString(app, { styles })

    // Epect the generated HTML to have generated styles and classes.
    expect(html).toMatchSnapshot()
  } catch (err) {
    fail(err)
  }
})
