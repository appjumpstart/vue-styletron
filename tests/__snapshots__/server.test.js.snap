// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`server generates styles and classes 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <title>Test Template</title>
    <style class=\\"_styletron_hydrate_\\">.ae{font-size:48px}.af{text-align:center}</style>
  <style class=\\"_styletron_hydrate_\\">.ae{font-size:48px}.af{text-align:center}</style></head>
  <body>
    <div data-server-rendered=\\"true\\" class=\\"ae af mr-5\\">
  Hello!
</div>
  </body>
</html>
"
`;
